test/specs/register-unicon-breeder.e2e.js

```diff
@@ -0,0 +1,160 @@
+import unicornsTextPage from '~/test/page-objects/unicorns-text.page.js'
+import noOfUnicornsStaffPage from '~/test/page-objects/no-of-unicorns-staff.page.js'
+import summaryPage from '~/test/page-objects/summary.page.js'
+
+describe('Register unicorn breeder form - e2e', () => {
+  before(async () => {
+    await nameEntryPage.open()
+  })
+
+  it('should enter name', async () => {
+    await expect(browser).toHaveTitle(`What's your name? - e2e form - GOV.UK`)
+    await nameEntryPage.enterName.setValue('John Doe')
+    await nameEntryPage.submitButton.click()
+
+    await expect(browser).toHaveTitle(
+      `What's your email address? - e2e form - GOV.UK`
+    )
+  })
+
+  it('should enter email address', async () => {
+    await emailPage.enterEmail.setValue('john.doe@example.co.uk')
+    await emailPage.submitButton.click()
+    await expect(browser).toHaveTitle(
+      `What's your phone number? - e2e form - GOV.UK`
+    )
+  })
+
+  it('should enter phone number', async () => {
+    await phoneNumberPage.enterPhoneNumber.setValue('07779875')
+    await phoneNumberPage.submitButton.click()
+
+    await expect(browser).toHaveTitle(
+      `What's your address? - e2e form - GOV.UK`
+    )
+  })
+
+  it('should enter address', async () => {
+    await Promise.all([
+      addressPage.enterAddressLine1.setValue('1 High Street'),
+      addressPage.enterTown.setValue('Townsville'),
+      addressPage.enterPostcode.setValue('TS1 1ST')
+    ])
+    await addressPage.submitButton.click()
+
+    await expect(browser).toHaveTitle(
+      'Do you want your unicorn breeder certificate sent to this address? - e2e form - GOV.UK'
+    )
+  })
+
+  it('should select Yes to use same address for certificate', async () => {
+    await certificateAddressPage.selectYesOrNo.click()
+    await certificateAddressPage.submitButton.click()
+
+    await expect(browser).toHaveTitle(
+      'When does your unicorn insurance policy start? - e2e form - GOV.UK'
+    )
+  })
+
+  it('should enter policy start date', async () => {
+    await Promise.all([
+      policyStartDatePage.enterDay.setValue('01'),
+      policyStartDatePage.enterMonth.setValue('02'),
+      policyStartDatePage.enterYear.setValue('2024')
+    ])
+
+    await policyStartDatePage.submitButton.click()
+
+    await expect(browser).toHaveTitle(
+      'Upload your insurance certificate - e2e form - GOV.UK'
+    )
+  })
+
+  it('should upload file', async () => {
+    const filePath = path.resolve('./test/file/test-file.txt')
+    await uploadFilePage.chooseFile.setValue(filePath)
+    await uploadFilePage.uploadFile.click()
+
+    let fileUploaded = true
+
+    while (fileUploaded) {
+      const inLink = uploadFilePage.refreshPage
+
+      await inLink.click()
+      const uploadedFileElem = uploadFilePage.checkForUploadedFile
+
+      if (await uploadedFileElem.isExisting()) {
+        fileUploaded = false
+      }
```

Comment:
Not something we need to worry about now, but longer term it'd be ideal to pull these generic utils out of the form-specific tests and into the page class so we can reuse it between different forms.